subworkflows/local/nonpareil.nf

```diff
@@ -0,0 +1,73 @@
+include { NONPAREIL_NONPAREIL        } from '../../modules/nf-core/nonpareil/nonpareil/main'
+include { NONPAREIL_CURVE            } from '../../modules/nf-core/nonpareil/curve/main'
+include { NONPAREIL_SET              } from '../../modules/nf-core/nonpareil/set/main'
+include { NONPAREIL_NONPAREILCURVESR } from '../../modules/nf-core/nonpareil/nonpareilcurvesr/main'
+
+// Custom Functions
+
+/*
+
+*/
+def extractNonpareilExtensionFromArrays(ch_input) {
+
+return ch_profile
+    .map { meta, profile -> [meta.db_name, meta, profile] }
+    .combine(ch_database, by: 0)
+    .multiMap {
+        key, meta, profile, db_meta, db ->
+            profile: [meta, profile]
+            db: db
+    }
+}
+
+workflow NONPAREIL {
+    take:
+    reads     // [ [ meta ], [ reads ] ]
+
+    main:
+    ch_versions       = Channel.empty()
+    ch_multiqc_files  = Channel.empty()
+
+    ch_reads_for_nonpareil = reads
+                                .map {
+                                    meta, reads ->
+                                        def reads_new = meta.single_end ? reads : reads[0]
+                                        def format = reads_new[0].getBaseName().split('\\.').last() in ['fasta', 'fna', 'fa', 'fas'] ? 'fasta' : 'fastq'
```

Comment:
I need to check this ,it could be that this doesn't work now that we accept gzipped files...